### PR TITLE
Add `host_error_timeout_seconds` field to `compute_instance` on beta provider

### PR DESCRIPTION
**Pull request by karolgorc**
-Added field to `google_compute_instance` and `google_compute_instance_template`
-Added tests
-Added doc changes
-Added data_source support

Do you want me to introduce plan-time validation of this field or should the API handle errors on this field?

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `host_error_timeout_seconds` to `google_compute_instance` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/d/compute_instance.html.markdown [hunk 1]
```diff
@@ -186,6 +186,8 @@ The following arguments are supported:
 
 <a name="nested_scheduling"></a>The `scheduling` block supports:
 
+* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection
```
Comment (NickElliot) [inline mmv1/third_party/terraform/website/docs/d/compute_instance.html.markdown:189]:
```suggestion
* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection.
```

File: mmv1/third_party/terraform/website/docs/d/compute_instance_template.html.markdown [hunk 2]
```diff
@@ -276,7 +276,9 @@ The `disk_encryption_key` block supports:
    groups will use as host systems. Read more on sole-tenant node creation
    [here](https://cloud.google.com/compute/docs/nodes/create-nodes).
    Structure [documented below](#nested_node_affinities).
-   
+
+* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection
```
Comment (NickElliot) [inline mmv1/third_party/terraform/website/docs/d/compute_instance_template.html.markdown:280]:
```suggestion
* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection.
```

File: mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown [hunk 3]
```diff
@@ -260,7 +260,9 @@ The `disk_encryption_key` block supports:
    groups will use as host systems. Read more on sole-tenant node creation
    [here](https://cloud.google.com/compute/docs/nodes/create-nodes).
    Structure [documented below](#nested_node_affinities).
-   
+
+* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection
```
Comment (NickElliot) [inline mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown:264]:
```suggestion
* `host_error_timeout_seconds` - [Beta](https://terraform.io/docs/providers/google/guides/provider_versions.html) Time in seconds for host error detection.
```

**Review by NickElliot — APPROVED** [review]
if the API receiving an improper value aborts the entire operation, then it should be fine to allow the API to handle it. The concern with Terraform validation would be adding a maintenance cost should the values involved change. The reason to have it handled in Terraform is if the error could result in a potential permadiff or mismatch in state between local and remote.

**Review by NickElliot — CHANGES_REQUESTED** [review]
misclick

**Comment by karolgorc** [comment]
Just to clarify. Yes the error-handling is fully supported on the API side. Any incorrect input will fail the apply so there is no risk of having a permadiff.

**Review by NickElliot — COMMENTED** [review]
Need to ask one last question: for the "default behavior" is `host_error_timeout_seconds` still being returned in GET requests, or is the field not returned at all?

**Comment by karolgorc** [comment]
While testing your question i found a bug that probably has something to do with the additional function for Diff on `scheduling` block. When removing the `host_error_timeout_seconds` field, terraform doesn't see a change in the config. Will work on a fix for this and write additional test

**Review by NickElliot — APPROVED** [review]
Given it's part of behavior for the full block I think it's fine to approve as is, and introduce a new PR for fixing the wider bug. Thanks for the contribution!

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_template_test.go.erb [hunk 4]
```diff
@@ -158,6 +158,44 @@ func TestAccComputeInstanceTemplate_maintenance_interval(t *testing.T) {
 }
 <% end -%>
 
+<% unless version == "ga" -%>
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_template_test.go.erb:161]:
```suggestion
<% unless version == 'ga' -%>
```

**Pull request merged**